### ISSUE TITLE
Service account for a shared cluster?

### ISSUE THREAD

**Issue by swapnild2111** [clastix/flux2-capsule-multi-tenancy#3]
Hello Team,
At this line - https://github.com/clastix/flux2-capsule-multi-tenancy/blob/4b15286f075d0b12675cefef4d316bc7bc30d35e/clusters/staging/flux-system/kustomization.yaml#L24

We are configuring tenant's reconciler SA, this is fine when we have a cluster per tenant scenario.

But how do we configure for a cluster - with multiple tenants sharing cluster & single flux installation on it?

**Comment by maxgio92** [comment]
Hi @swapnild2111,
thank you for your feedback.

I'd like to ask you a question to understand your scenario. You have a cluster shared between tenants, which are managed by Capsule, isn't it?

**Comment by swapnild2111** [comment]
Yes we have a shared cluster where multiple tenants deploy their applications. We mainly use Flux to deploy.
Now we want to introduce tight policies in the cluster with Capsule.

We also customer/product specific clusters, where Capsule can we with this flag as there is single SA which flux can use.

But am not sure how to make this work in shared cluster.

**Comment by maxgio92** [comment]
Perfect @swapnild2111, thank you.

Actually, the documentation considers your scenario of a shared cluster. So, you don't need to do anything different :)

This line:
https://github.com/clastix/flux2-capsule-multi-tenancy/blob/4b15286f075d0b12675cefef4d316bc7bc30d35e/clusters/staging/flux-system/kustomization.yaml#L24
comes from the official [documentation of Flux](https://fluxcd.io/flux/security/best-practices/#multi-tenancy-lock-down) for multi-tenant scenarios like this one, leveraging the lock-down features.

It exactly says that by default, a `ServiceAccount` named _gitops-reconciler_, will be used by the _kustomize-controller_ to reconcile `Kustomization`s via user impersonation, when not otherwise explicited a different `ServiceAccount` at [`Kustomization`-level](https://fluxcd.io/flux/components/kustomize/api/v1/#kustomize.toolkit.fluxcd.io/v1.KustomizationSpec).

Either way (explicited or enforced the default one), that `ServiceAccount`, must exists in the same `Namespace` of the `Kustomization`, and it will be used by the controller to build a client and make requests impersonating the related Kubernetes user (i.e. named `serviceaccount:<namespace>:<serviceaccount_name>`). 

My recommendation is to read here: https://fluxcd.io/flux/security/best-practices/#multi-tenancy-lock-down. It's well described both the setup and the rationale behind it, alongside other security recommendations.

If you have other questions or need support, let me know :-)

**Comment by swapnild2111** [comment]
Thank you for reply :) I figured out what I was doing wrong - we can close this :)

**Comment by maxgio92** [comment]
Thanks to you @swapnild2111. May I ask you to share your use case here?
It could be useful for the community, and also to improve the documentation :-)

**Comment by prometherion** [comment]
Couldn't agree more with @maxgio92, please, @swapnild2111: contributing to the open source is also sharing use cases to inspire other developers or organizations.

If you're concerned about NDA or sensitive data, please, reach out to us at [#capsule](https://kubernetes.slack.com/archives/C03GETTJQRL) Kubernetes Slack Workspace channel.
